src/input_handlers.h

```diff
@@ -119,68 +119,28 @@ class RecvFromInputHandler : public MessageParser<InPlaceAccumulation> {
     }
 };
 
-#ifdef USING_VMA_EXTRA_API
-class SocketXtremeInputHandler : public MessageParser<BufferAccumulation> {
-private:
-    SocketRecvData &m_recv_data;
-    vma_buff_t *m_vma_buff;
-public:
-    inline SocketXtremeInputHandler(Message *msg, SocketRecvData &recv_data):
-        MessageParser<BufferAccumulation>(msg),
-        m_recv_data(recv_data),
-        m_vma_buff(NULL)
-    {}
-
-    /** Receive pending data from a socket
-     * @param [in] socket descriptor
-     * @param [out] recvfrom_addr address to save peer address into
-     * @param [inout] in - storage size, out - actual address size
-     * @return status code
-     */
-    inline int receive_pending_data(int fd, struct sockaddr *recvfrom_addr, socklen_t &size)
-    {
-        //TODO: update after IPv6 will be supported in libvma
-        size = sizeof(sockaddr_in);
-        std::memcpy(recvfrom_addr, &g_vma_comps->src, size);
-        m_vma_buff = g_vma_buff;
-        if (likely(m_vma_buff)) {
-            return m_vma_buff->len;
-        } else {
-            return 0;
-        }
-    }
-
-    template <class Callback>
-    inline bool iterate_over_buffers(Callback &callback)
-    {
-        for (vma_buff_t *cur = m_vma_buff; cur; cur = cur->next) {
-            bool res = process_buffer(callback, m_recv_data, (uint8_t *)cur->payload, cur->len);
-            if (unlikely(!res)) {
-                return false;
-            }
-        }
-        return true;
-    }
-
-    inline void cleanup()
-    {
-    }
-};
+#if defined(USING_VMA_EXTRA_API) || defined(USING_XLIO_EXTRA_API)
+typedef int(* recvfrom_zcopy_func_t)(int __fd, void *__buf, size_t __nbytes, int *__flags,
+                                     struct sockaddr *__from, socklen_t *__fromlen);
 
-class VmaZCopyReadInputHandler : public MessageParser<BufferAccumulation> {
-private:
+template <typename PACKET, typename PACKETS, int MSG_ZCOPY_FLAG>
```

Comment:
In most coding styles template parameters starts with capital letter letter and ALL_CAPITAL are reserved for constants.
```suggestion
template <typename Packet, typename PacketsDataType, int MsgZCopyFlag>
```

Reply:
Done